integrations/zendesk/src/definitions/channels.ts

```diff
@@ -0,0 +1,36 @@
+import type { IntegrationDefinitionProps } from '@botpress/sdk'
+import { z } from 'zod'
+
+export const channels = {
+  ticket: {
+    title: 'Zendesk Ticket',
+    messages: {
+      text: {
+        schema: z.object({
+          text: z.string(),
+          userId: z.string(),
```

Comment:
**chore:** user id has nothing to do in the message payload. It should be a tag on the user. You bot will receive it in the message handler

update: I just saw that there is already an `id` tag on the user, so use that instead

Reply:
Yeah but the thing is that the user in `props.user.id` is not the same as in `props.payload.userId`. 

When a ticket is created, we create a new conversation and a user that is associated with it, and it's returned so it can be used subsequently for the duration of the conversation with the agent. In the dialog manager, we override the conversationId and the userId to send the message at the correct destination. 

Not sure if it should be done differently, but that's the working solution we ended up with @slvnperron

Reply:
I don't understand what you are trying to explain, but you don't have to \@ tag sly to convince me man (it does not work on me)

The value in `props.payload.userId` is the **botpress** user id. I assumed the one in the message payload was for the **zendesk** user id ? If it's not, what is it ? 

Do you want to have a call ?

Reply:
I tagged him because i'm pretty sure he can explain it better than me lol, it works but i'm not sure if it's the best approach

Reply:
Ok I understand, I resolve the comment and start another review